src/rust-2021/prelude.md

```diff
@@ -31,5 +30,85 @@ It's identical to the current one, except for three new additions:
 - [`std::convert::TryFrom`](https://doc.rust-lang.org/stable/std/convert/trait.TryFrom.html)
 - [`std::iter::FromIterator`](https://doc.rust-lang.org/stable/std/iter/trait.FromIterator.html)
 
-The library team still needs to formally approve these, which will likely happen soon.
-<!-- TODO: hopefully this happens before we publish this -->
+The tracking issue [can be found here](https://github.com/rust-lang/rust/issues/85684).
+
+## Migration to Rust 2021
+
+As a part of the 2021 edition a migration lint, `rust_2021_prelude_collisions`, has been added in order to aid in automatic migration of Rust 2018 codebases to Rust 2021.
+
+In order to have rustfix migrate your code to be Rust 2021 Edition compatible, run:
+
+```ignore
+cargo fix --edition
+```
+
+The lint detects cases where functions or methods are called that have the same name as the methods defined in one of the new prelude traits. In some cases, it may rewrite your calls in various ways to ensure that you continue to call the same function you did before.
+
+### Fully qualified calls to inherent methods
+
+Many types define their own inherent methods with the name `from_iter` which shares the same name with `std::iter::FromIterator::from_iter`:
+
+```rust
+struct MyStruct {
+  data: Vec<u32>
+}
+
+impl MyStruct {
+  // This has the same name as `std::iter::FromIterator::from_iter`
+  fn from_iter(x: impl Iterator<Item = u32>) -> Self {
+    Self {
+      data: x.collect()
+    }
+  }
+}
+```
+
+In case that already use a fully qualified inherent method syntax (e.g., calls like `MyStruct::from_iter`), there is not ambiguity with calls to trait methods (e.g., `<MyStruct as FromIter>::from_iter`), because inherent methods take precedence over trait methods. Therefore, no migration is needed.
+
+A migration is necessary when using "dot method" syntax where the method name is the same as the trait method name for a trait which is now
+in scope. For example, a call like `my_struct.into_iter()` where `into_iter()` used to refer to an inherent method on `MyStruct` but now conflicts with
+the trait method from `IntoIter`. To make these calls unambiguous, fully qualified inherent method syntax must be used:
```

Comment:
This paragraph is a little confusing to me.  My understanding is that inherent method calls should *not* require migration, since they take precedence over trait methods. I'm also a little confused since this shifts gears and is talking about `IntoIter` which isn't new for 2021. Is there a test or playground example showing how this would be a migration?

Reply:
Yes, you're right. this was based on my misunderstanding of when the migration was necessary. I'll clean this up. Also good call to not use `IntoIter`.

Reply:
@ehuss made the change, can you take a look again?